reference/gall.md

```diff
@@ -0,0 +1,765 @@
+unsubscribed, but they also could be closed by an intermediary.  For
+example, if a subscription is from another ship which is currently
+unreachable, Ames may choose to close the subscription to avoid queueing
+updates indefinitely.  If the program crashes while processing an
+update, this may also generate an unsubscription.  You should consider
+subscriptions to be closable at any time.
+
+#### Accepts
+
+```hoon
+=path
+```
+
+Path of the closed subscription.
+
+#### Returns
+
+```hoon
+(quip card _this)
+```
+
+List of cards and new agent.
+
+#### Example
+
+```hoon
+:: From reference agent. Prints a message when programs become unsubscribed.
+++  on-leave
+  |=  =path
+  ^-  (quip card _this)
+  ~&  "Unsubscribe by: {<src.bowl>} on: {<path>}"
+  `this
+```
+
+### `++on-peek`
+
+Handle scry request.
+
+This arm is called when a program reads from the agent's "scry"
+namespace, which should be referentially transparent.  Unlike most
+handlers, this cannot perform IO, and it cannot change the state.  All
+it can do is produce a piece of data to the caller, or not.
+
+#### Accepts
+
+```hoon
+=path
+```
+
+The path being scryed for.
+
+#### Returns
+
+```hoon
+(unit (unit cage))
+```
+
+If this arm produces `[~ ~ data]`, then `data` is the value at the the
+given path.  If it produces `[~ ~]`, then there is no data at the given
+path and never will be.  If it produces `~`, then we don't know yet whether
+there is or will be data at the given path. The head of the path is known as the
+`care`. Requests with a care of `%x` should return a vase that matches the mark
+at the end of the path. Requests with a care of `%y` should return a vase with
+an `arch` inside.
+
+#### Example
+
+```hoon
+:: From reference agent. Allows for scrying for local and remote counters as well
+:: as a list of remote counters being viewed.
+++  on-peek
+  |=  =path
+  ^-  (unit (unit cage))
+  ?+  path  [~ ~]
+      ::
+    [%x %local ~]
+  ``[%atom !>(local.sta)]
+      ::
+    [%x %remote who=@ta ~]
+  ~&  remote.sta
+  =/  res  (~(got by remote.sta) (slav %p -.+.+.path))
+  ~&  res
+  ``[%atom !>(res)]
+      ::
+    [%y %remote ~]
+  =/  dir=(map @ta ~)
+    (molt (turn ~(tap by remote.sta) |=([who=@p *] [(scot %p who) ~])))
+  ``[%arch !>(`arch`[~ dir])]
+  ==
+::
+```
+
+### `++on-agent`
+
+Handle `%pass` card
+
+This arm is called to handle responses to `%pass` cards to other agents.
+It will be one of the following types of response:
+
+- `%poke-ack`: acknowledgment (positive or negative) of a poke.  If the
+  value is `~`, then the poke succeeded.  If the value is `[~ tang]`,
+  then the poke failed, and a printable explanation (eg a stack trace)
+  is given in the `tang`.
+
+- `%watch-ack`: acknowledgment (positive or negative) of a subscription.
+  If negative, the subscription is already ended (technically, it never
+  started).
+
+- `%fact`: update from the publisher.
+
+- `%kick`: notification that the subscription has ended.
```

Comment:
Worth noting that this happens either because of the target app killing the subscription, or ames doing it due to pressure. Most of the time, you want to try resubscribing. If you were kicked, and it's no longer valid to access, you'll get a negative `%watch-ack` and end your flow there.